internal/cmd/validate_test.go

```diff
@@ -0,0 +1,174 @@
+package cmd
+
+import (
+	"context"
+	_ "embed"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/logr/testr"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"gotest.tools/v3/assert"
+
+	"package-operator.run/package-operator/internal/packages/packagecontent"
+)
+
+func TestValidateConfig(t *testing.T) {
+	t.Parallel()
+
+	testLogger := testr.New(t)
+
+	for name, tc := range map[string]struct {
+		Options  []ValidateOption
+		Expected ValidateConfig
+	}{
+		"defaults": {
+			Expected: ValidateConfig{
+				Log:    logr.Discard(),
+				Puller: &defaultPackagePuller{},
+			},
+		},
+		"with logger": {
+			Options: []ValidateOption{
+				WithLog{Log: testLogger},
+			},
+			Expected: ValidateConfig{
+				Log:    testLogger,
+				Puller: &defaultPackagePuller{},
+			},
+		},
+	} {
+		tc := tc
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var cfg ValidateConfig
+
+			cfg.Option(tc.Options...)
+			cfg.Default()
+
+			assert.Equal(t, tc.Expected, cfg)
+		})
+	}
+}
+
+func TestValidatePackageConfig(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		Options   []ValidatePackageOption
+		Assertion require.ErrorAssertionFunc
+	}{
+		"no options": {
+			Options:   []ValidatePackageOption{},
+			Assertion: require.Error,
+		},
+		"empty path": {
+			Options: []ValidatePackageOption{
+				WithPath(""),
+			},
+			Assertion: require.Error,
+		},
+		"empty remote reference": {
+			Options: []ValidatePackageOption{
+				WithRemoteReference(""),
+			},
+			Assertion: require.Error,
+		},
+		"mutually exclusive options": {
+			Options: []ValidatePackageOption{
+				WithPath("test"),
+				WithRemoteReference("test"),
+			},
+			Assertion: require.Error,
+		},
+	} {
+		tc := tc
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var cfg ValidatePackageConfig
+
+			cfg.Option(tc.Options...)
+
+			tc.Assertion(t, cfg.Validate())
+		})
+	}
+}
+
+//go:embed testdata/manifest.yaml
+var _manifest []byte
```

Comment:
Why is this prefixed with an underscore?

Reply:
I follow this advice https://github.com/uber-go/guide/blob/master/style.md#prefix-unexported-globals-with-_. Helps me avoid mistakes when using global state, but totally up to preference.